### PR TITLE
addressing sonarlint issues

### PR DESCRIPTION
**Pull request by ryancrawcour**
## Type of PR

- [ ] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
major refactor to address the issues raised by sonarlint.

## Review notes
still 3 issues left at the moment, that i do not yet know how to resolve. 
will address those with another PR in the future once we've discussed and found a solution. 

## Issues Closed or Referenced
- References #183

### CODE REVIEW

File: src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/sink/CosmosDBSinkTask.java [hunk 1]
```diff
@@ -52,30 +52,30 @@ public void put(Collection<SinkRecord> records) {
...
-                    logger.debug("Value.toString(): " + (record.value() != null ? record.value().toString() : "<null>"));
-                }
+                logger.debug("Writing record, value type: {}", record.value().getClass().getName());
+                logger.debug("Key Schema: {}", record.keySchema());
+                logger.debug("Value schema: {}", record.valueSchema());
+                logger.debug("Value.toString(): {}",  record.value() != null ? record.value().toString() : "<null>");
+                
                 try {
-                    if(!this.settings.getUseUpsert()) {
+                    if (Boolean.TRUE.equals(this.settings.getUseUpsert())) {
```
Comment (kev-ms) [inline src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/sink/CosmosDBSinkTask.java:74]: Use `equalsIgnoreCase` instead if used this way

**Review by kev-ms — COMMENTED** [review]
looks good to me

**Comment by ryancrawcour** [comment]
Won’t getUseUpsert return a Boolean, and therefore case is irrelevant?

**Comment by kev-ms** [comment]
yes, my bad for some reason I thought it was some constants comparison..

File: src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/sink/CosmosDBSinkTask.java [hunk 2]
```diff
@@ -52,30 +52,30 @@ public void put(Collection<SinkRecord> records) {
...
-        logger.info("Sending " + records.size() + " records to be written");
+        logger.info("Sending {} records to be written", records.size());
 
         Map<String, List<SinkRecord>> recordsByContainer = records.stream()
                 //Find target collection for each record
                 .collect(Collectors.groupingBy(record ->
                         settings.getTopicContainerMap().getContainerForTopic(record.topic()).orElseThrow(
-                                () -> new IllegalStateException("No container defined for topic " + record.topic() + "."))));
+                                () -> new IllegalStateException(String.format("No container defined for topic %s .",
+                                record.topic())))));
```
Comment (sivamu) [inline src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/sink/CosmosDBSinkTask.java:62]: Why not use the `{}` notation here? Or does it not work for the record.topic()?

File: src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/LinkedTransferQueue.class [hunk 3]
```diff
@@ -0,0 +1,1749 @@
+/*
```
Comment (sivamu) [inline src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/LinkedTransferQueue.class:1]: I don't think this class is needed

File: src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/sink/CosmosDBSinkConnector.java [hunk 4]
```diff
@@ -33,34 +30,40 @@ public void start(Map<String, String> props) {
...
+        logger.debug("config invoked");
+
         ConfigDef configDef = new ConfigDef();
         new SinkSettings().getAllSettings().stream().forEachOrdered(setting -> setting.toConfigDef(configDef));
-        logger.debug("Sink ConfigDef with " + configDef.configKeys().size() + " settings.");
+
+        logger.debug(
+            "Sink ConfigDef with {} settings",
+            configDef.configKeys().size()
+        );
```
Comment (sivamu) [inline src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/sink/CosmosDBSinkConnector.java:57]: Nit but can this be a one-liner?

File: src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/LinkedTransferQueue.class (hunk 3 above)
Comment (ryancrawcour) [reply src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/LinkedTransferQueue.class:1]: image

it is used in the Source Connector as its building up the internal buffer.

**Review by sivamu — APPROVED** [review]

File: src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/sink/CosmosDBSinkTask.java (hunk 2 above)
Comment (ryancrawcour) [reply src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/sink/CosmosDBSinkTask.java:62]: as discussed, won't fix this.

File: src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/LinkedTransferQueue.class (hunk 3 above)
Comment (sivamu) [reply src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/LinkedTransferQueue.class:1]: But if it's importing from java.util.concurrent, why do we still need to make our own? I'm confused

Comment (ryancrawcour) [reply src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/LinkedTransferQueue.class:1]: good point. i don't really know why this one was being used vs something from java.util.concurrent. i just know the class is being used. 

we can change the code to use native java data structures if these will suffice and give us the same functionality.

**Pull request merged**